caikit/core/model_management/model_trainer_base.py

```diff
@@ -28,15 +28,22 @@
 # Standard
 from typing import Optional, Type
 import abc
+import dataclasses
 import os
 
 # Local
-from ..data_model import TrainingInfo
+from ..data_model import TrainingStatus
 from ..modules import ModuleBase
 from ..toolkit.factory import FactoryConstructible
 from ..toolkit.reversible_hasher import ReversibleHasher
 
 
+@dataclasses.dataclass
+class TrainingInfo:
+    status: TrainingStatus
+    errors: Optional[Exception] = None
```

Comment:
@gkumbhat do we want a list or a single one here? We should choose and make the plural-ness of the noun here match.

I think the only reason we came up with to go with a list of exceptions was for the case of data validation where we may want to have a big old list of exceptions for each spot where there's a problem with the user's data.

```
@dataclasses.dataclass
class TrainingInfo:
    status: TrainingStatus
    exception: Optional[Exception] = None
    # or
    exceptions: List[Exception]  # can be empty
```